examples/core/transform-feedback-instanced/app.js

```diff
@@ -0,0 +1,406 @@
+
+      float cos_r = cos(a_rotation);
+      float sin_r = sin(a_rotation);
+      mat2 rot = mat2(
+          cos_r, sin_r,
+          -sin_r, cos_r
+      );
+      gl_Position = vec4(rot * a_position + a_offset, 0.0, 1.0);
+
+      // -hack-
+      // gl_Position = vec4(a_position, 0.0, 1.0);
+      // v_color.xy = a_offset;
+      // v_color.z = a_rotation;
+}
+`;
+
+const FS_DRAW = `#version 300 es
+  #define ALPHA 0.9
+
+  precision highp float;
+  precision highp int;
+
+  in vec3 v_color;
+
+  out vec4 color;
+
+  void main()
+  {
+      color = vec4(v_color * ALPHA, ALPHA);
+}
+`;
+
+const NUM_INSTANCES = 1000;
+let currentSourceIdx = 0;
+
+const animationLoop = new AnimationLoop({
+  glOptions: {webgl2: true},
+  // eslint-disable-next-line
+  onInitialize({canvas, gl}) {
+
+    const modelRender = new Model(gl, {
+      id: 'Model-Render',
+      program: new Program(gl, {
+        vs: VS_DRAW,
+        fs: FS_DRAW
+      }),
+      geometry: new Geometry({
+        id: 'Model-Render-Geometry',
+        drawMode: gl.TRIANGLE_FAN,
+        vertexCount: 3
+      }),
+      isInstanced: true,
+      instanceCount: NUM_INSTANCES,
+      isIndexed: false
+    });
+
+    const modelTransform = new Model(gl, {
+      id: 'Model-Transform',
+      program: new Program(gl, {
+        vs: VS_EMIT,
+        fs: FS_EMIT,
+        varyings: VARYINGS
+      }),
+      geometry: new Geometry({
+        id: 'Model-Transform-Geometry',
+        drawMode: gl.POINTS,
+        vertexCount: NUM_INSTANCES
+      }),
+      isInstanced: false,
+      isIndexed: false
+    });
+
+    // -- Initialize data
+    const trianglePositions = new Float32Array([
+      0.015, 0.0,
+      -0.010, 0.010,
+      -0.010, -0.010
+    ]);
+
+    const instanceOffsets = new Float32Array(NUM_INSTANCES * 2);
+    const instanceRotations = new Float32Array(NUM_INSTANCES);
+    const instanceColors = new Float32Array(NUM_INSTANCES * 3);
+
+    for (let i = 0; i < NUM_INSTANCES; ++i) {
+      const oi = i * 2;
+      const ri = i;
+      const ci = i * 3;
+
+      instanceOffsets[oi] = Math.random() * 2.0 - 1.0;
+      instanceOffsets[oi + 1] = Math.random() * 2.0 - 1.0;
+
+      instanceRotations[ri] = Math.random() * 2 * Math.PI;
+
+      instanceColors[ci] = Math.random();
+      instanceColors[ci + 1] = Math.random();
+      instanceColors[ci + 2] = Math.random();
+    }
+
+    // -- Init VertexArrays and TransformFeedback objects
+
+    const vertexArrays = new Array(2);
+    const transformVertexArrays = new Array(2);
+    const transformFeedbacks = new Array(2);
+
+    const positionBuffer = new Buffer(gl, {
+      data: trianglePositions,
+      size: 2,
+      type: gl.FLOAT,
+      normalized: false,
+      stride: 0,
+      offset: 0,
+      instanced: 0,
+      usage: gl.STATIC_DRAW
+    });
+
+    const colorBuffer = new Buffer(gl, {
+      data: instanceColors,
+      size: 3,
+      type: gl.FLOAT,
+      normalized: false,
+      stride: 0,
```

Comment:
This is quite verbose, doesn't show the luma.gl API in the best light...

Don't we have defaults on these so we don't have to specify all? 

Or are we making a point of specifying them all here in the example?